src/features/solutions/getAllSolutionsSlice.js

```diff
@@ -1,25 +1,49 @@
 import { createSlice, createAsyncThunk } from "@reduxjs/toolkit";
-import http from "../../axiosInstance"
+import solutionService from "../../services/solution.service";
 
-export const getAllSolutions = createAsyncThunk("solutions/get", async (query) => {
-    try {
-        const response = await http.get(`/solutions`);
-        return response;
-    } catch (err) {
-        if (!err.response) {
-            throw err;
+
+const initialState = {
+    solutions: [],
+    status: "idle",
+    error: null
+}
+
+export const getAllSolutions = createAsyncThunk(
+    "solutions/get",
+    async () => {
+        try {
+            const response = await solutionService.getAll();
+            return response.data?.data;
+        } catch (err) {
+            if (!err.response) {
+                throw err;
+            }
+            Promise.reject(err.response);
         }
-        Promise.reject(err.response);
-    }
-})
+    })
 
 export const getAllSolutionsSlice = createSlice({
-    name: "getAllSolutions",
-    initialState: {},
+    name: "solutions",
+    initialState,
     reducers: {},
-    extraReducers: () => {},
+    extraReducers: (builder) => {
+        builder
+            .addCase(getAllSolutions.pending, (state) => {
+                state.status = 'loading';
+            })
+            .addCase(getAllSolutions.fulfilled, (state, action) => {
+                state.status = "succeeded";
+                state.solutions = action.payload;
+            })
+            .addCase(getAllSolutions.rejected, (state, action) => {
+                state.status = 'failed';
+                state.error = action.error.message;
+            })
+    },
```

Comment:
- Status should not be a string
- Refactor your `initialState` based on below example
```
initialState: {
    books: [],
    loading: false,
    error: null,
  },
```